### PR TITLE
[DDW-1069] Rotating logs

### PR DESCRIPTION
**Pull request by Sam-Jeston**
This PR makes sure Daedalus log packing includes cardano-node log.

Implementation summary:
 - Rotate logs with the use of rotating-file-stream and stdio pipe of child process
 - Move node logs into pub directory
 - Update ALLOWED_NODE_LOGS regex

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/messages/GGKFXSKC6)
- [ ] Test

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

File: source/main/cardano/CardanoWalletLauncher.js [hunk 1]
```diff
@@ -43,15 +43,20 @@ export async function CardanoWalletLauncher(
       break;
   }
 
-  const walletStdio: string[] = ['inherit', logStream, logStream, 'ipc'];
+  const walletStdio: string[] = ['pipe', 'pipe', 'pipe', 'ipc'];
```
Comment (cleverca22) [inline source/main/cardano/CardanoWalletLauncher.js]: what happens if you just leave this line as-is, i believe its already the right type to work here

Comment (Sam-Jeston) [reply source/main/cardano/CardanoWalletLauncher.js]: It should have been, as the return type of the library was an extension of `WriteStream`, but when passed to `stdio` it threw an invalid type error, saying that it was not a valid Stream.

I couldn't see any difference in the CPU profile when using pipe, so I think its okay.

Comment (Sam-Jeston) [reply source/main/cardano/CardanoWalletLauncher.js]: And to confirm, traditionally with a Stream you wait for the `on` event as that represents when the fd is opened, at which point it is valid to be passed to `stdio`. However that didn't help, and `on` was called every time the log was cycled, which was super annoying as then we tried to launch new wallets and nodes each log cycle!

So this change actually solves more than one thing

File: source/main/config.js [hunk 2]
```diff
@@ -104,7 +104,7 @@ export const ALLOWED_LOGS = [
   'Daedalus-versions.json',
   'State-snapshot.json',
 ];
-export const ALLOWED_NODE_LOGS = new RegExp(/(node.json-)(\d{14}$)/);
+export const ALLOWED_NODE_LOGS = new RegExp(/node.log-/);
```
Comment (CodiePP) [inline source/main/config.js]: the new node will have the timestamp in between its log name "node" and the extension ".json". 
example: `node-20191113060708.json`
this made it easier for support to read the files, as Windows recognises the file extension.

File: source/main/cardano/CardanoNode.js [hunk 3]
```diff
@@ -275,48 +275,61 @@ export class CardanoNode {
...
-        } catch (_) {
-          reject(
-            new Error('CardanoNode#start: Error while spawning cardano-node')
-          );
+          return `node.log-${Date.now()}`;
+        },
+        {
+          size: '5M',
+          path: config.logFilePath,
+          maxFiles: 10,
```
Comment (DeeJayElly) [inline source/main/cardano/CardanoNode.js]: I think we need to set the limit of maxFiles to 4. Right @nikolaglumac ?

Comment (nikolaglumac) [reply source/main/cardano/CardanoNode.js]: No this is actually fine - we need to pack only 4 latest logs into our logs-download. Can you please make sure it works like that @DeeJayElly?

File: source/main/config.js (hunk 2 above)
Comment (nikolaglumac) [reply source/main/config.js]: We should leave it as is for now as this is also how Daedalus logs are named...

**Comment by DmitriiGaico** [comment]
@Sam-Jeston Hi Sam! Could you please advise what exactly file name for the Cardano Log file?
What i see is node.log-current file and it is packed in Zip.  Thank you in advance.

**Pull request merged**